contracts/contracts/strategies/NativeStaking/ValidatorRegistrator.sol

```diff
@@ -0,0 +1,341 @@
+        _;
+    }
+
+    /// @dev Throws if called by any account other than the Strategist
+    modifier onlyStrategist() {
+        require(
+            msg.sender == IVault(VAULT_ADDRESS).strategistAddr(),
+            "Caller is not the Strategist"
+        );
+        _;
+    }
+
+    /// @param _wethAddress Address of the Erc20 WETH Token contract
+    /// @param _vaultAddress Address of the Vault
+    /// @param _beaconChainDepositContract Address of the beacon chain deposit contract
+    /// @param _ssvNetwork Address of the SSV Network contract
+    constructor(
+        address _wethAddress,
+        address _vaultAddress,
+        address _beaconChainDepositContract,
+        address _ssvNetwork
+    ) {
+        WETH_TOKEN_ADDRESS = _wethAddress;
+        BEACON_CHAIN_DEPOSIT_CONTRACT = _beaconChainDepositContract;
+        SSV_NETWORK_ADDRESS = _ssvNetwork;
+        VAULT_ADDRESS = _vaultAddress;
+    }
+
+    /// @notice Set the address of the registrator which can register, exit and remove validators
+    function setRegistrator(address _address) external onlyGovernor {
+        emit RegistratorChanged(_address);
+        validatorRegistrator = _address;
+    }
+
+    /// @notice Set the address of the staking monitor that is allowed to reset stakeETHTally
+    function setStakingMonitor(address _address) external onlyGovernor {
+        emit StakingMonitorChanged(_address);
+        stakingMonitor = _address;
+    }
+
+    /// @notice Set the amount of ETH that can be staked before staking monitor
+    // needs to a approve further staking by resetting the stake ETH tally
+    function setStakeETHThreshold(uint256 _amount) external onlyGovernor {
+        emit StakeETHThresholdChanged(_amount);
+        stakeETHThreshold = _amount;
+    }
+
+    /// @notice Reset the stakeETHTally
+    function resetStakeETHTally() external onlyStakingMonitor {
+        emit StakeETHTallyReset();
+        stakeETHTally = 0;
+    }
+
+    /// @notice Stakes WETH to the node validators
+    /// @param validators A list of validator data needed to stake.
+    /// The `ValidatorStakeData` struct contains the pubkey, signature and depositDataRoot.
+    /// Only the registrator can call this function.
+    // slither-disable-start reentrancy-eth
+    function stakeEth(ValidatorStakeData[] calldata validators)
+        external
+        onlyRegistrator
+        whenNotPaused
+    {
+        uint256 requiredETH = validators.length * 32 ether;
+
+        // Check there is enough WETH from the deposits sitting in this strategy contract
+        require(
+            requiredETH <= IWETH9(WETH_TOKEN_ADDRESS).balanceOf(address(this)),
+            "insufficient WETH"
+        );
+
+        require(
+            stakeETHTally + requiredETH <= stakeETHThreshold,
+            "Staking ETH over threshold"
+        );
+        stakeETHTally += requiredETH;
+
+        // Convert required ETH from WETH
+        IWETH9(WETH_TOKEN_ADDRESS).withdraw(requiredETH);
```

Comment:
A payable receive function seems to be missing to get the ETH out of the WETH withdraw.

Reply:
The `receive` is on the `NativeStakingSSVStrategy` contract as it also need to accept ETH from the FeeAccumulator.

I could change to use inheritance so it's clearer. That is, implement a `receive` on `ValidatorRegistrator` to accept payments from WETH and then override with a `super.receive` call on `NativeStakingSSVStrategy` to also accept payments from the `FeeAccumulator`. Thoughts?

Reply:
The inheritance idea with super will not work as that will execute the two require statements separately. Payment needs to be allowed from WETH or the FeeAccumulator.
The `receive` on `NativeStakingSSVStrategy` would have to override the `receive` on `ValidatorRegistrator` with no `super` call.